### PR TITLE
Adapt protein mutation setup pipeline to handle glycosylated RBD:ACE2

### PR DESCRIPTION
**Pull request by zhang-ivy**
This is a temporary fix that allows generation of htfs for glycosylated RBD:ACE2, since the amber glycan forcefield is not currently compatible with OpenMM. 

**Overview of the setup pipeline for glycosylated RBD:ACE2:**
1. Obtain apo RBD and apo ACE2 PDBs (does not need glycan bonding to be there). Edit the PDBs for tleap if necessary 
2. Solvate and parametrize 1) apo RBD and 2) RBD:ACE2 in tleap
3. Load prmtop and inpcrd files from 2) into OpenMM and correct chain/residue ids in topology
*Output: old topology (corrected prmtop), old positions (inpcrd), old system (prmtop)*
4. Load the tleap output prmtop/inpcrd from 2) into MDAnalysis and correct chain/residue ids and remove solvent —> save as PDB
5. Mutate PDB from 4) using pymol mutate.py
6. Load mutated PDB and renumber the atoms based on the atom ordering in new_topology. Then copy the solvent positions from the old positions —> save as mutated and solvated PDB
7. Correct the mutated and solvated PDB for tleap 
8. Parametrize in tleap (do not solvate)
*Output: new topology (created from old topology), new positions (inpcrd), new system (prmtop)*

**Summary of changes**
In `rjmc/topology_proposal.py`:
- Created `PointMutationEngineRBD` (subclass of `PointMutationEngine`)
- Sped up `_construct_atom_map`

In `app/relative_point_mutation_setup.py`: 
- Created `PointMutationExecutorRBD` (subclass of `PointMutationExecutor`) 

Created `utils/rbd.py`:
- Contains utility functions used in `PointMutationEngineRBD` and `PointMutationExecutor`

TO DO:

- [ ] Create /data/rbd_ace2/ dir and add template tleap.in and mutate.py (+ other necessary scripts/files)
- [ ] Add examples to docstrings

### CODE REVIEW

File: perses/rjmc/topology_proposal.py [hunk 1]
```diff
@@ -2043,6 +2047,7 @@ def _construct_atom_map(self,
         new_oemol_res_copy : openeye.oechem.oemol object
             copy of modified new oemol
         """
+        _logger.debug("In construct atom map")
```
Comment (zhang-ivy) [inline perses/rjmc/topology_proposal.py:2050]: @zhang-ivy delete this

**Comment by jchodera** [comment]
Just noting that we don't want to merge this PR, and instead want to finish the [work converting th glycam force field converted to OpenMM's `ffxml`](https://github.com/openmm/openmmforcefields/issues/150) in the short term.

Longer term, the [Open Force Field Initiative](http://openforcefield.org) will provide fully self-consistent general force fields that work for glycosylated residues transparently.

**Comment by mikemhenry** [comment]
Since https://github.com/openmm/openmmforcefields/issues/150 got merged, do we need this anymore @zhang-ivy @jchodera ?

**Comment by zhang-ivy** [comment]
Closing this as we've converted the glycam ff to an ffxml, so this hack is no longer necessary

**Pull request closed without merging**